mod_dynamic_spawns/classes/unit_block.nut

```diff
@@ -114,14 +114,26 @@
 		}
 		else
 		{
-			local function satisfiesTierWidth( _index )
+			local spawnables = this.__DynamicSpawnables.filter(@(_, _unit) _unit.canSpawn());
+			if (spawnables.len() == 0)
+				return null;
+
+			// Returns true if all the units at and above "my index + this.TierWidth" have no units spawned yet.
+			// Because we check units from index 0 onwards and if any higher unit beyond the TierWidth already
+			// has units spawned, then we should not spawn this unit as it would violate the TierWidth.
+			local function satisfiesTierWidth( _idx )
```

Comment:
What about the tiers below _idx? They could also cause the tierwidth to be violated

Reply:
We check the tiers from 0 onwards so whether a lower tier violates the tierWidth gets naturally checked due to that logic already.

Reply:
But nothing happens when a lower tier violates the tierwidth. The foreach loop just keeps going

Reply:
It is prevented from spawning though. A lower tier which violates the TierWidth can only be present if it got spawned during a cycle when the higher tiers' `canSpawn` was false. Therefore, it shouldn't continue to spawn, instead it should be skipped and the higher tier should be spawned. See my [reply](https://github.com/Battle-Modders/Dynamic-Spawns-Framework/pull/17#discussion_r2043662649) in the other conversation.